SW/main/Thermal.py

```diff
@@ -24,46 +25,126 @@ def __init__(self):
         
         #initialize the sensor
         self.sensor = Adafruit_AMG88xx()
-        
+
+        self.points = [(math.floor(ix / 8), (ix % 8)) for ix in range(0, 64)]
+        self.grid_x, self.grid_y = np.mgrid[0:7:32j, 0:7:32j]
+
+    @staticmethod
+    def map(x, in_min, in_max, out_min, out_max):
+      return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min
+
+    def get_temperature(self, img_in, img_out, pixels_in):
+
+        lst_intensities = []
+        areas = []
+        i = 0       
+        avg = 0.0
+
+        #turn input image into grayscale
+        blurred_in = cv2.GaussianBlur(img_in, (7, 7), 0)
+        im_in = np.stack((blurred_in,) * 3,-1)
+        im_in = im_in.astype(np.uint8)
+        bgr_in = cv2.cvtColor(im_in, cv2.COLOR_RGB2BGR)
+        gray_in = cv2.cvtColor(bgr_in, cv2.COLOR_BGR2GRAY)
+
+	#turn output image into grayscale
```

Comment:
having a hard time following -- add comments to explain why you're taking these steps